vcd/datasource_vcd_lb_app_profile.go

```diff
@@ -0,0 +1,113 @@
+				Optional:    true,
+				ForceNew:    true,
+				Description: "vCD organization in which the App Profile is located",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "vCD virtual datacenter in which the App Profile is located",
+			},
+			"edge_gateway": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway name in which the App Profile is located",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "App Profile name for lookup",
+			},
+			"type": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+				Description: "Protocol type used to send requests to the server. One of 'TCP', " +
+					"'UDP', 'HTTP' org 'HTTPS'",
+			},
+			"enable_ssl_passthrough": &schema.Schema{
+				Type:     schema.TypeBool,
+				Computed: true,
+				Description: "Enable SSL authentication to be passed through to the virtual " +
+					"server. Otherwise SSL authentication takes place at the destination address.",
+			},
+			"http_redirect_url": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+				Description: "The URL to which traffic that arrives at the destination address " +
+					"should be redirected. Only applies for types HTTP and HTTPS",
+			},
+			"persistence_mechanism": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+				Description: "Persistence mechanism for the profile. One of 'cookie', " +
+					"'ssl-sessionid', 'sourceip'",
+			},
+			"cookie_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+				Description: "Used to uniquely identify the session the first time a client " +
+					"accesses the site. The load balancer refers to this cookie when connecting " +
+					"subsequent requests in the session, so that they all go to the same virtual " +
+					"server. Only applies for persistence_mechanism 'cookie'",
+			},
+			"cookie_mode": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+				Description: "The mode by which the cookie should be inserted. One of 'insert', " +
+					"'prefix', or 'appsession'",
+			},
+			"expiration": &schema.Schema{
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Length of time in seconds that persistence stays in effect",
+			},
+			"insert_x_forwarded_http_header": &schema.Schema{
+				Type:     schema.TypeBool,
+				Computed: true,
+				Description: "Enables 'X-Forwarded-For' header for identifying the originating IP" +
+					" address of a client connecting to a Web server through the load balancer. " +
+					"Only applies for types HTTP and HTTPS",
+			},
+			// TODO https://github.com/terraform-providers/terraform-provider-vcd/issues/258
+			// This will not give much use without SSL certs being available
+			"enable_pool_side_ssl": &schema.Schema{
```

Comment:
only when updating after manually uploaded? - So maybe documentation and description should state that! Even Warning outputted? @lvirbalas

Reply:
Added more comments here and in docs. Could throw a warning output if we agree so.

Reply:
Looks good to me. Warning would be useful for those who use it expecting more, but it would be annoying for those who set the things manually, wouldn't it?

Reply:
Yes, makes sense - too much noise. Also - whoever does it and understands it they will immediately spot that there are missing fields to point to certificates (we can add them once we have certificate handling).